pkg/macaroon/apikey.go

```diff
@@ -131,6 +131,27 @@ func (a *APIKey) Check(ctx context.Context, secret []byte, action Action, revoke
 	return nil
 }
 
+// GetAllowedBuckets returns a list of all the allowed bucket paths that match the Action operation
+func (a *APIKey) GetAllowedBuckets(ctx context.Context, action Action) (allowedBuckets map[string]struct{}, err error) {
```

Comment:
@zeebo is there a better way to handle this?

Reply:
No, I don't see a better way to do this. Did you have any specific concerns?

Reply:
Just seemed something that could look nicer.